### DISCUSSION TITLE
[Clickhouse] No Field Class

### DISCUSSION THREAD

**Discussion by asafcombo** [apache/superset#18490] [Ideas]
When selecting on a table with fields of non standard type, a 

`No Field Class of <TYPE>`   error is displayed.

This happens with 'simple' types such as tuples  or with more complex types as:

`SimpleAggregateFunction(sum, UInt64)`

or 

`AggregateFunction(uniq, String)`

Currently the workaround is to manually CAST the field into its proper type. 

A more elegant solution would be that superset would cast it by the type automatically. 

So for a column such as 

```
    `count`                             SimpleAggregateFunction(sum, UInt64) 
```

superset should understand the type is Uint64 and do 

`CAST(count as Uint64)`

For tuples the only method I found was to cast it as String

related to https://github.com/apache/superset/issues/14445

**Comment by Slach** [comment]
look like `latest` superset + latest clickhouse-sqlalchemy already resolve `issue`

![image](https://user-images.githubusercontent.com/105560/129240388-7bde5194-c762-487a-bbc0-ffce07bfb247.png)

@asafcombo could you check 1.2.0 version with your data
https://gist.github.com/Slach/ad926dfa30ceb4fe1d168c00c3da98e0?

**Comment by asafcombo** [comment]
I can confirm that according to my testing the latest "git+https://github.com/xzkostyan/clickhouse-sqlalchemy.git"  is sufficient to fix the error.

**Comment by Slach** [comment]
@junlincc could we close this issue?
